### PR TITLE
feat: configuration priority ENV > emqx.conf > API

### PR DESCRIPTION
**Pull request by zhongwencool**
Fixes <issue-or-jira-number>

1. Change the priority of the configuration: `ENV > emqx.conf > API`.
2. Deprecated `data/configs/cluster-override.conf` and `data/configs/local-override.conf`.
3. Stabilizing the HTTP API for hot updates.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [x] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` and `.zh.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [x] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/src/emqx_config.erl [hunk 1]
```diff
@@ -390,11 +407,21 @@ parse_hocon(Conf) ->
...
         true -> hocon:binary(Conf, Opts);
         false -> hocon:files(Conf, Opts)
+    end;
+do_parse_hocon(false, Conf, IncDirs) ->
+    Opts = #{format => map, include_dirs => IncDirs},
+    case is_binary(Conf) of
+        true ->
+            hocon:binary(Conf, Opts);
+        false ->
+            LocalFile = deprecated_override_conf_file(#{override_to => local}),
```
Comment (zhongwencool) [inline apps/emqx/src/emqx_config.erl:422]: delete local.hocon

File: apps/emqx/src/emqx_config.erl [hunk 2]
```diff
@@ -480,17 +507,36 @@ ensure_file_deleted(F) ->
...
-    File = override_conf_file(Opts),
+    File =
+        case has_deprecated_file() of
+            true -> deprecated_conf_file(Opts);
+            false -> cluster_hocon_file(Opts)
+        end,
     load_hocon_file(File, map).
 
-override_conf_file(Opts) when is_map(Opts) ->
+has_deprecated_file() ->
```
Comment (zmstone) [inline apps/emqx/src/emqx_config.erl:517]:
```suggestion
%% @doc Return `true' if this node is upgraded from older version which used cluster-override.conf for
%% cluster-wide config persistence.
has_deprecated_file() ->
```

File: apps/emqx/src/emqx_config.erl [hunk 3]
```diff
@@ -480,17 +507,36 @@ ensure_file_deleted(F) ->
...
         case maps:get(override_to, Opts, cluster) of
             local -> local_override_conf_file;
             cluster -> cluster_override_conf_file
         end,
     application:get_env(emqx, Key, undefined);
-override_conf_file(Which) when is_atom(Which) ->
+deprecated_conf_file(Which) when is_atom(Which) ->
+    application:get_env(emqx, Which, undefined).
+
+cluster_hocon_file(Opts) when is_map(Opts) ->
```
Comment (zmstone) [inline apps/emqx/src/emqx_config.erl:531]:
```suggestion
%% The newer version cluster-wide config persistence file.
cluster_hocon_file(Opts) when is_map(Opts) ->
```

File: apps/emqx/src/emqx_config.erl [hunk 4]
```diff
@@ -480,17 +507,36 @@ ensure_file_deleted(F) ->
...
+    Key =
+        case maps:get(override_to, Opts, cluster) of
```
Comment (zmstone) [inline apps/emqx/src/emqx_config.erl:533]: there is no longer a need for this `cluster` option.

**Review by zmstone — COMMENTED** [review]
Please add a changelog.

File: changes/ce/feat-10156.en.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+Change the priority of the configuration:
+1. If it is a new installation of EMQX, the priority of s configuration is `ENV > emqx.conf > HTTP API`.
```
Comment (HJianBo) [inline changes/ce/feat-10156.en.md:2]:
```suggestion
1. If it is a new installation of EMQX, the priority of configuration is `ENV > emqx.conf > HTTP API`.
```

File: changes/ce/feat-10156.en.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+2. If it is an upgrade of EMQX, the priority of s configuration is the same as before `HTTP API > ENV > emqx.conf`.
```
Comment (HJianBo) [inline changes/ce/feat-10156.en.md:3]:
```suggestion
2. If EMQX is upgraded from an old version (i.e., the cluster-override.conf file still exists in EMQX's data directory), then the configuration priority remains the same as before. That is, `HTTP API > ENV > emqx.conf`.
```

**Review by HJianBo — APPROVED** [review]

**Pull request merged**